### ISSUE TITLE
regression: attributes double on rustfmt

### ISSUE THREAD

**Issue by behnam** [rust-lang-nursery/rustfmt#1674]
I just did updated to latest `rustfmt` and now running it on files with attributes duplicate the attributes; not sure if all of them, but at least some of them.

Here's a repro using test files on `master`:
```
$ rustfmt tests/target/attrib.rs
$ git diff
diff --git a/tests/target/attrib.rs b/tests/target/attrib.rs
index ed152a8..0c69e23 100644
--- a/tests/target/attrib.rs
+++ b/tests/target/attrib.rs
@@ -52,10 +52,11 @@ struct Foo {
 // #1668

 /// Default path (*nix)
-#[cfg(all(unix, not(target_os = "macos"), not(target_os = "ios"), not(target_os = "android")))]
+#[cfg(all(unix, not(target_os = "macos") ), not(target_os = "ios") ),
+            not(target_os = "android")))]
 fn foo() {
     #[cfg(target_os = "freertos")]
-    match port_id {
+    #[cfg(target_os = "freertos")] match port_id {
         'a' | 'A' => GpioPort { port_address: GPIO_A },
         'b' | 'B' => GpioPort { port_address: GPIO_B },
         _ => panic!(),
```

**Comment by nrc** [comment]
This looks like a dup of #1668 which was fixed by #1669, but perhaps it is a related bug, not the same one. Could you check if your master branch includes the fix please?

**Comment by behnam** [comment]
Yeah, I cannot repro on dev builds on `master` NOR tag `0.8.6`. But repros from `rustfmt:0.8.6` installed by `cargo install`. Anyways, looks like a dup, so I'm closing.
